src/ValueObject/Request/Request.php

```diff
@@ -0,0 +1,165 @@
+<?php
+/**
+ * Copyright Shopgate Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @author    Shopgate Inc, 804 Congress Ave, Austin, Texas 78701 <interfaces@shopgate.com>
+ * @copyright Shopgate Inc
+ * @license   http://www.apache.org/licenses/LICENSE-2.0 Apache License, Version 2.0
+ */
+
+namespace Shopgate\CloudIntegrationSdk\ValueObject\Request;
+
+class Request
+{
+    /** @var string */
+    private $uri;
+
+    /** @var string */
+    private $method;
+
+    /** @var string[] */
+    private $headers;
+
+    /** @var string */
+    private $body;
+
+    /**
+     * Request constructor.
+     *
+     * @param string        $uri
+     * @param string        $method
+     * @param string[]      $headers
+     * @param string | null $body
+     */
+    public function __construct($uri, $method, $headers = array(), $body = null)
```

Comment:
can be more strict ```array $headers = ```

Reply:
good idea. but i will also keep the = array() as default value, for easier use